pedpy/methods/acceleration_calculator.py

```diff
@@ -0,0 +1,346 @@
+"""Module containing functions to compute accelerations."""
+
+from typing import Optional, Tuple
+
+import numpy as np
+import numpy.typing as npt
+import pandas
+import shapely
+
+from pedpy.column_identifier import A_X_COL, A_Y_COL, ACC_COL, FRAME_COL, ID_COL
+from pedpy.data.geometry import MeasurementArea
+from pedpy.data.trajectory_data import TrajectoryData
+from pedpy.methods.method_utils import (
+    AccelerationCalculation,
+    _compute_individual_movement_acceleration,
+)
+
+
+class AccelerationError(Exception):
+    """Class reflecting errors when computing accelerations with PedPy."""
+
+    def __init__(self, message):
+        """Create AccelerationError with the given message.
+
+        Args:
+            message: Error message
+        """
+        self.message = message
+
+
+def compute_individual_acceleration(
+    *,
+    traj_data: TrajectoryData,
+    frame_step: int,
+    movement_direction: Optional[npt.NDArray[np.float64]] = None,
+    compute_acceleration_components: bool = False,
+    acceleration_calculation: AccelerationCalculation = AccelerationCalculation.BORDER_EXCLUDE,
+) -> pandas.DataFrame:
+    r"""Compute the individual acceleration for each pedestrian.
+
+    For computing the individuals' acceleration at a specific frame :math:`a_i(t_k)`,
+    a specific frame step (:math:`n`) is needed.
+    Together with the
+    :attr:`~trajectory_data.TrajectoryData.frame_rate` of
+    the trajectory data :math:`fps` the time frame :math:`\Delta t` for
+    computing the speed becomes:
+
+    .. math::
+
+        \Delta t = 2 n / fps
+
+    This time step describes how many frames before and after the current
+    position :math:`X(t_k)` are used to compute the movement.
+    These positions are called :math:`X(t_{k+n})`, :math:`X(t_{k-n})`
+    respectively.
+
+    In order to compute the acceleration at time 't_k', we first calculate the
+    displacements :math:`\bar{X}` around 't_{k+n}' and 't_{k-n}':
+
+    .. math::
+
+        \bar{X}(t_{k+n}) = X(t_{k+2n}) - X(t_{k})
+
+    .. math::
+
+        \bar{X}(t_{k-n}) = X(t_{k}) - X(t_{k-2n})
+
+    The acceleration is then calculated from the difference of the displacements
+
+    .. math::
+
+        \Delta\bar{X}(t_k) = \bar{X}(t_{k+n}) - \bar{X}(t_{k-n})
+
+    divided by the square of the time interval '\Delta t':
+
+    .. math::
+
+        a_i(t_k) = \Delta\bar{X}(t_k) / \Delta t**2
```

Comment:
This should be "a_i(t_k) = \Delta\bar{X}(t_k) / \Delta t^{2}"

Reply:
done